File: 2079-watering-plants/2079-watering-plants.kt
class Solution {
    fun wateringPlants(plants: IntArray, capacity: Int): Int {
        var bucket = capacity
        var steps = 0
        for (i in 0..plants.size-1) {
            if(bucket >= plants[i]){
                bucket -= plants[i]
                steps++
            }
            else{
                steps += (2*i) + 1
                bucket = capacity
                bucket -= plants[i]
            }
        }
        return steps
    }
}

File: 2079-watering-plants/2079-watering-plants.cpp
class Solution {
public:
    int wateringPlants(vector<int>& plants, int capacity) {
        int bucket = capacity;
        int steps = 0;
        for(int i = 0; i < plants.size(); i++){
            if(bucket >= plants[i]){
                bucket -= plants[i];
                steps++;
            }
            else{
                steps += (2*i) + 1;
                bucket = capacity;
                bucket -= plants[i];
            }
        }
        return steps;
    }
};